internal/middleware/authentication.go

```diff
@@ -67,60 +68,38 @@ func (m AuthenticationMiddleware) TokenAuthentication(c *gin.Context) {
 		c.Abort()
 		return
 	} else {
-		c.Set("user", u)
+		c.Set("user", user)
 		c.Next()
 	}
 }
 
 func parseRequest(request *http.Request, key *rsa.PublicKey) (*model.User, error) {
 	token, err := jwt.ParseRequest(
 		request,
-		jwt.WithValidate(true),
-		jwt.WithVerify(jwa.RS256, key),
+		jwt.WithKey(jwa.RS256, key),
+		jwt.WithHeaderKey("Authorization"),
+		jwt.WithCookieKey("accessToken"),
+		jwt.WithCookieKey("refreshToken"),
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	userData, ok := token.Get("user")
-	if !ok {
-		return nil, errors.New("user not found in claims")
-	}
-
-	return extractUser(userData)
+	return extractUser(token)
 }
 
-func extractUser(userData any) (*model.User, error) {
-	userMap, ok := userData.(map[string]any)
+func extractUser(token jwt.Token) (*model.User, error) {
+	userData, ok := token.Get("user")
 	if !ok {
-		return nil, errors.New("failed to parse user data")
+		return nil, errors.New("user not found in claims")
 	}
 
-	id := userMap["id"].(float64)
-	email := userMap["email"].(string)
-
-	user := &model.User{
-		ID:          uint(id),
-		Email:       email,
-		Groups:      extractGroups("groups", userMap),
-		AdminGroups: extractGroups("adminGroups", userMap),
+	bytes, err := json.Marshal(userData)
```

Comment:
Why do we serialize to Json and deserialize it right away?

Reply:
I am not exactly sure what `userData` is. Is it a JSON string? Or does is it actually a `*model.User`? In the latter case we can type assert. If its not maybe https://github.com/lestrrat-go/jwx/pull/391 can help 🤔

Reply:
Yes! Good find!